backend/app/api/endpoints/devices.py

```
@@ -120,3 +160,151 @@ async def delete_device(
+                detail="Failed to send upgrade command to device",
+            )
+
+    except Exception as e:
+        logger.exception(
+            f"[Device Upgrade] Error triggering upgrade: "
+            f"user_id={user_id}, device_id={device_id}, error={e}"
+        )
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to trigger upgrade: {str(e)}",
+        )
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)